addOns/llm/src/main/java/org/zaproxy/addon/llm/services/LlmCommunicationService.java

```diff
@@ -0,0 +1,179 @@
+import dev.langchain4j.service.AiServices;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.httpclient.util.HttpURLConnection;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpSender;
+import org.zaproxy.addon.llm.communication.Confidence;
+import org.zaproxy.addon.llm.communication.HttpRequestList;
+import org.zaproxy.addon.llm.ui.settings.LlmOptionsParam;
+import org.zaproxy.addon.llm.utils.HistoryPersister;
+import org.zaproxy.addon.llm.utils.Requestor;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+
+public class LlmCommunicationService {
+
+    private static final Logger LOGGER = LogManager.getLogger(LlmCommunicationService.class);
+    private static final String AI_REVIEWD_TAG_KEY = "AI-Reviewed";
+
+    private LlmAssistant llmAssistant;
+    private LlmResponseHandler listener;
+    Requestor requestor;
+
+    static ChatLanguageModel model;
+    ChatMemory chatMemory;
+
+    public LlmCommunicationService(String modelName, String apiKey, String endpoint) {
+        listener = new LlmResponseHandler();
+        chatMemory = MessageWindowChatMemory.withMaxMessages(10);
+        model =
+                AzureOpenAiChatModel.builder()
+                        .apiKey(apiKey)
+                        .deploymentName(modelName)
+                        .endpoint(endpoint)
+                        .temperature(0.3)
+                        .responseFormat(new ChatCompletionsJsonResponseFormat())
+                        .listeners(List.of(listener))
+                        .logRequestsAndResponses(true)
+                        .build();
+
+        llmAssistant =
+                AiServices.builder(LlmAssistant.class)
+                        .chatLanguageModel(model)
+                        .chatMemory(chatMemory)
+                        .build();
+        requestor = new Requestor(HttpSender.MANUAL_REQUEST_INITIATOR, new HistoryPersister());
+    }
+
+    private Integer importHttpCalls(String swaggercontent) throws IOException {
+        HttpRequestList listHttpRequest = llmAssistant.extractHttpRequests(swaggercontent);
+        if (listHttpRequest == null) {
+            throw new RuntimeException("An issue occurred when trying to get response from the LLM");
+        }
+        requestor.run(listHttpRequest);
+        return listHttpRequest.getRequests().size();
+    }
+
+    public Integer importOpenapiFromUrl(String urlString) {
+        Integer endpointCount = 0;
+        try {
+            URL url = new URL(urlString);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```

Comment:
Should use ZAP classes rather than Java ones for comms. Users may well have configured an outgoing proxy in ZAP which the Java classes won't know about.
The OpenAPI add-on does this: [Requestor.java](https://github.com/zaproxy/zap-extensions/blob/main/addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/network/Requestor.java)